public/app/plugins/panel/piechart/PieChartPanel.tsx

```diff
@@ -29,7 +30,7 @@ export const PieChartPanel: React.FC<Props> = ({
       width={width}
       height={height}
       timeZone={timeZone}
-      eventBus={eventBus}
+      eventBus={new EventBusWithSource(eventBus, id.toString(10))}
```

Comment:
would be nice not to have to create a new event bus for every render, maybe can live in state?

Reply:
Ah, good point